src/uwtools/set_namelist.py

```diff
@@ -0,0 +1,124 @@
+    parser.add_argument('-o', '--outfile',
+                       help='Full path to output file. This is a \
+                       namelist by default.',
+                       required=False,
+                       )
+    # Jinja2 user input file as a template to moc f90 name list file
+    parser.add_argument('-i', '--template_input_nml',
+                       required=True,
+                       help='Path to a templated namelist.',
+                       )
+    # YAML config file for name list generation containing name list values
+    parser.add_argument('-c', '--config',
+                       required=False,
+                       help='Path ito yaml configuration file for setting f90 name list generation',
+                       )
+    # single switch for printing to stdout the f90 namelist output file
+    parser.add_argument('-d', '--dry_run',
+                       action='store_true',
+                       help='If provided, suppress writing out file and redirect to stdout.',
+                       )
+
+    # switch for printing to stdout the required values needed to filled in by the template
+    parser.add_argument('--values_needed',
+                       action='store_true',
+                       help='If provided, suppress all output.',
+                       )
+
+
+    parser.add_argument("--set",
+                        metavar="KEY=VALUE",
+                        nargs='*',
+                        help="Set a number of key-value pairs "
+                             "(do not put spaces before or after the = sign). "
+                        )
+
+    return parser.parse_args(argv)
+
+def get_nl_values(cla):
+    '''get name list values to be filled designated by command line'''
+
+    def file_exists_and_readable(check_file):
+        if os.path.isfile(check_file):
+            if not os.access(check_file, os.R_OK):
+                print(f"File {check_file} exists but is is not readable")
+                sys.exit(-1)
+        else:
+            print(f"File {check_file} does not exist")
+            sys.exit(-1)
+
+    file_exists_and_readable(cla.template_input_nml)
+    if cla.config is not None:
+        file_exists_and_readable(cla.config)
+
+        nl_values_yaml = config.YAMLConfig(config_path=cla.config)
+        nl_values_yaml.parse_include(config_file=cla.config,from_environment=True)
+        if cla.set is not None:
+            nl_commandline_values = dict(map(lambda s: s.split('='), cla.set))
+            nl_values_yaml.parse_include(data=nl_commandline_values)
+
+    if cla.config is None and cla.set is not None:
+        nl_values_yaml = dict(map(lambda s: s.split('='), cla.set))
+
+    if cla.config is None and cla.set is None:
+        print("No values for templating name list were given. Defaultng to environment variables only") #pylint: disable=line-too-long
+        nl_values_yaml = os.environ.copy()
+
+    return nl_values_yaml
+
+
+def set_namelist(argv):
```

Comment:
Not sure what the relationship of `tempater.py` to "Name List" tool is.

Reply:
@TerrenceMcGuinness-NOAA My apologies if this confusion arrived from my misspelling of `templater.py`.

If that is not the source of misunderstanding, here goes an attempt to clarify...during the course of PI 5, the design tasks for a namelist tool and a model_configure tool led to the path forward that we'd have 2 tools that would handle all our configuration file types.  You can find the design docs on our Confluence page, and they will be on the GitHub Wiki soon. 

The first of the two tools is the `templater.py` which is meant to read in any flat file formatted as a Jinja2 template, without a need to know anything about its syntax, use, etc. and render the template based on a user's configuration. This will essentially replace tools like `sed` and `atparse` to unify around Jinja2.

The second of the two tools will address our "dictable" file types -- yaml, fortran namelists, INI files, etc. In our design docs, I named this one `set_config.py`, but I'm open to other more useful/descriptive names. This tool will allow a user to make any change to any key/value pair that they may wish to change given a configuration file that has desired key/value pairs. This tool will be a bit more powerful, also providing the ability to convert between different configuration file types (namelist to YAML, for example), and to compare two unordered files of the same type.

Reply:
Ok, this all was never made clear to me and was a total loss when the `set_config.py` came up.  I suspect this discussion is the is reflexive of description "As **_we_** have re-imagined the use of this tool ..." in [UW-204](https://jira-epic.woc.noaa.gov/browse/UW-204).